src/sync.ts

```diff
@@ -602,7 +602,7 @@ export class SyncApi {
             if (shouldClear) {
                 this.storeIsInvalid = true;
                 const reason = InvalidStoreError.TOGGLED_LAZY_LOADING;
-                const error = new InvalidStoreError(reason, !!this.opts.lazyLoadMembers);
+                const error = new InvalidStoreError(reason, !!this.opts.lazyLoadMembers) as any;
```

Comment:
need a comment to explain why we're casting to `any`, or better yet: fix the error that prevents the casting from working.